src/common/Core/Search/SearchFilter.ts

```diff
@@ -0,0 +1,15 @@
+import { SearchResultItem } from "../SearchResultItem";
+import { fuseJsSearchFilter } from "./fuseJsSearchFilter";
+import { fuzzySortFilter } from "./fuzzySortFilter";
+import { SearchOptions } from "./SearchOptions";
+
+const searchFilters = {
+    "Fuse.js": (options: SearchOptions) => fuseJsSearchFilter(options),
+    "fuzzysort": (options: SearchOptions) => fuzzySortFilter(options),
+};
+
+export type SearchEngineId = keyof typeof searchFilters;
+
+export function searchFilter(options: SearchOptions & { searchEngineId: SearchEngineId }): SearchResultItem[] {
+    return searchFilters[options.searchEngineId](options);
+}
```

Comment:
Please don't use union types like this just for convenience. Split this into 2 parameters:

```suggestion
export const searchFilter = (options: SearchOptions, searchEngineId: SearchEngineId): SearchResultItem[] =>
    searchFilters[searchEngineId](options);
```

Reply:
Fixed and added eslint rule